pkg/controllers/management/etcdbackupconfig/etcdbackupconfig.go

```diff
@@ -0,0 +1,179 @@
+				c.kickBackup(bc.Name)
+				continue
+			}
+			lastCompletedTime, _ := time.Parse(time.RFC3339, lastCompleted.LastUpdateTime)
+			// It's been more than bc.Creation since we ran last successfull backup
+			if time.Since(lastCompletedTime) > configDuration {
+				c.kickBackup(bc.Name)
+			}
+		}
+	}
+
+}
+
+func (c *Controller) setSeenCondition(backupConfigName string) error {
+	bc, _ := c.getBackupConfig(backupConfigName)
+	setCondition(bc, v3.EtcdBackupConfigCondition{
+		Type:           "LastSeen",
+		Status:         "True",
+		LastUpdateTime: time.Now().Format(time.RFC3339),
+	})
+
+	return c.updateBackupConfig(bc)
+}
+
+func (c *Controller) setLastStartedCondition(backupConfigName string) error {
+	bc, err := c.getBackupConfig(backupConfigName)
+	if err != nil {
+		return err
+	}
+	setCondition(bc, v3.EtcdBackupConfigCondition{
+		Type:           "LastStarted",
+		Status:         "True",
+		LastUpdateTime: time.Now().Format(time.RFC3339),
+	})
+
+	return c.updateBackupConfig(bc)
+}
+
+func (c *Controller) setLastCompletedConddition(backupConfigName string) error {
+	bc, err := c.getBackupConfig(backupConfigName)
+	if err != nil {
+		return err
+	}
+	setCondition(bc, v3.EtcdBackupConfigCondition{
+		Type:           "LastCompleted",
+		Status:         "True",
+		LastUpdateTime: time.Now().Format(time.RFC3339),
+	})
+
+	return c.updateBackupConfig(bc)
+}
+
+func (c *Controller) getBackupConfig(name string) (*v3.EtcdBackupConfig, error) {
+	return c.backupConfigClient.Get(name, metav1.GetOptions{})
+}
+
+func (c *Controller) updateBackupConfig(bc *v3.EtcdBackupConfig) error {
+	_, err := c.backupConfigClient.Update(bc)
+	return err
+}
+
+func setCondition(backupConfig *v3.EtcdBackupConfig, newCondition v3.EtcdBackupConfigCondition) {
+	conditions := []v3.EtcdBackupConfigCondition{}
+	for _, c := range backupConfig.Status.Conditions {
+		if c.Type == newCondition.Type {
+			continue
+		}
+		conditions = append(conditions, c)
+	}
+	backupConfig.Status.Conditions = append(conditions, newCondition)
+}
+
+func getCondition(backupConfig *v3.EtcdBackupConfig, cType string) *v3.EtcdBackupConfigCondition {
+	for _, c := range backupConfig.Status.Conditions {
+		if c.Type == cType {
+			return &c
+		}
+	}
+	return nil
+}
+
+func (c *Controller) kickBackup(backupConfigName string) {
+	logrus.Infof("melsayed doing the backup")
+	backupConfig, _ := c.getBackupConfig(backupConfigName)
+	newBackup := &v3.EtcdBackup{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-", backupConfigName),
+		},
+		BackupConfig: *backupConfig,
+		Status: v3.EtcdBackupStatus{
+			Conditions: []v3.EtcdBackupCondition{
+				v3.EtcdBackupCondition{
+					Type:           "Created",
+					Status:         v1.ConditionTrue,
+					LastUpdateTime: time.Now().Format(time.RFC3339),
+				},
+			},
+		},
+	}
+	fmt.Printf("%#v\n", newBackup)
+	_, err := c.backupClient.Create(newBackup)
+	if err != nil {
+		logrus.Errorf("melsayed failed to cteate backup: %v", err)
```

Comment:
should wrap the c.backupClient.Create(newBackup) call in the condition framework call method; that will set the condition value to true if the backup was successful, and false if the backup has failed